docs/assets/js/bibtex-unicorn.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * Edit to the original bibtex_js_draw function
+ * Bibtex files need to be entered into individual <textarea>'s with class "reference"
+ * Generated content replaces the inputted bibtex data in each <textarea>
+ * Allows user to write notes/subheadings in between references
+ */
+$( document ).ready( function bibtex_js_draw_unicorn() {
+  var refs = document.querySelectorAll(".reference");
+  refs.forEach(ref => {
+    (new BibtexDisplay()).displayBibtex($(ref).val(), $("#bibtex_display") );
+    $(ref).replaceWith( $("#bibtex_display").html() );
+  });
+  document.querySelector(".references").style.display = "block"; // display page content when finished
+});
+
+/**
+ * Function to toggle the display of each bibtex abstract
+ */
+function toggleRef(element) {
+  var abstractElement = element.querySelector(".abstract");
```

Comment:
This will fail if there is no match for the selector, right?

Reply:
Do you mean if the reference doesn't have an abstract? In that case, this function won't ever be called, since the function call is nested within the ```<span class="if abstract">```. Bibtex-js removes the the entire snippet below if the "if" class isn't fulfilled. So there will always be a match for the selector.

Here's the snippet from the bibtex_template in references.html:
```html
<span class="if abstract">
  <span class="toggle" onclick="toggleAbstract(this)">[abstract]
    <span class="abstract" style="display: none;"></span>
  </span>
</span>
```

Reply:
True. But better be [safe than sorry](https://en.wikipedia.org/wiki/Defensive_programming), I guess? :smile: Other than that it's good to go!